### PR TITLE
Allow moving pieces in the UI

### PR DESCRIPTION
**Pull request by DanielAlbl**

### CODE REVIEW

File: src/board.rs [hunk 1]
```diff
@@ -91,6 +95,7 @@ impl Board {
             }
         }
         self.squares[to.y][to.x] = piece;
+        self.squares[from.y][from.x] = None;
```
Comment (Dillon-Roller) [inline src/board.rs:98]: why are we doing this? the code before does the exact same thing but it's shorter. `take()` leaves `None` inside

File: src/chess_widget.rs [hunk 2]
```diff
@@ -61,28 +108,85 @@ impl ChessWidget {
...
+            x: mouse_up.x - x_offset,
+            y: mouse_up.y - y_offset,
+        })
+    }
 }
 
 impl Widget<String> for ChessWidget {
-    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut String, _env: &Env) {}
+    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut String, _env: &Env) {
+        match _event {
```
Comment (Dillon-Roller) [inline src/chess_widget.rs:161]: for `_event` the `_` usually means allow it to be unused without warning, but it's being used here in the match, so should probably remove right?

File: src/chess_widget.rs [hunk 3]
```diff
@@ -61,28 +108,85 @@ impl ChessWidget {
...
+                        .ok();
+                }
+                self.mouse_down = None;
+            }
+            Event::MouseMove(event) => {
+                self.current_point = event.pos;
+                // if we are currently holding onto a piece, request a redraw
+                if let Some(mouse_down) = self.mouse_down {
+                    if self.game.get_piece(Position::from(mouse_down)).is_some() {
+                        _ctx.request_paint();
```
Comment (Dillon-Roller) [inline src/chess_widget.rs:183]: `_ctx` here too

File: src/board.rs (hunk 1 above)
Comment (DanielAlbl) [reply src/board.rs:98]: It was just doing `take` on a copy, it wasn't changing the one in `squares`.

File: src/chess_widget.rs (hunk 2 above)
Comment (DanielAlbl) [reply src/chess_widget.rs:161]: Lol I didn't even know that, just copied the function definition. I'll change it.

File: src/board.rs [hunk 4]
```diff
@@ -29,11 +29,15 @@ impl Board {
...
         squares[BOARD_SIZE - 1] = Self::get_back_rank(Player::Black);
 
-        Self {
+        let mut board = Self {
             squares,
             moves: HashSet::new(),
             player: Player::White,
-        }
+        };
+
```
Comment (Dillon-Roller) [inline src/board.rs:37]: So I would put code on the line right after `};` Also, we usually return without a blank line above it, but what do you think

Comment (DanielAlbl) [reply src/board.rs]: Yeah I agree

**Review by Dillon-Roller — APPROVED** [review]

**Pull request merged**